### DISCUSSION TITLE
Unable to Set Codespaces User Level Secret with GH CLI

### DISCUSSION THREAD

**Discussion by KyleJamesWalker** [community/community#16056] [Codespaces]
We are getting the following error when trying to set a Codespaces user level secret:

```
failed to fetch public key: HTTP 403: Must have admin rights to Repository. (https://api.github.com/user/codespaces/secrets/public-key)
```

The help command says `gh secret set --help` lists `gh secret set MYSECRET --user` should set a user-level secret for Codespaces, but when run we get the aforementioned error. We're assuming the `gh auth login` command is missing permissions.

Has anyone else had this issue, or is this something that needs to be fixed in the tool itself?

**Comment by KyleJamesWalker** [comment]
Note: If you manually create a token that has access to codespaces secrets you can set `GITHUB_TOKEN` env, and it will work, so that's at least a work around while we wait for this to get fixed.

**Comment by maxprokopiev** [comment]
did you try running `gh auth refresh -s codespace` to fix the permission scope?

**Reply by gm3dmo** [reply]
The `gh auth refresh -s codespace` codespace tip worked nicely for me. The error message seen was:

```
error getting codespaces: HTTP 403: Must have admin rights to Repository. (https://api.github.com/user/codespaces?per_page=30)
```


Thank you @maxprokopiev
